tmk_core/common/chibios/printf.c

```diff
@@ -211,19 +211,23 @@ void init_printf(void* putp, void (*putf)(void*, char)) {
     stdout_putp = putp;
 }
 
-void tfp_printf(char* fmt, ...) {
+int tfp_printf(const char* fmt, ...) {
     va_list va;
     va_start(va, fmt);
     tfp_format(stdout_putp, stdout_putf, fmt, va);
     va_end(va);
+
+    return 1;
 }
 
 static void putcp(void* p, char c) { *(*((char**)p))++ = c; }
 
-void tfp_sprintf(char* s, char* fmt, ...) {
+int tfp_sprintf(char* s, const char* fmt, ...) {
     va_list va;
     va_start(va, fmt);
     tfp_format(&s, putcp, fmt, va);
     putcp(&s, 0);
     va_end(va);
+
+    return 1;
```

Comment:
Was a void before, so no one can be checking the return code....

Question is, while not perfect, will this do?

Reply:
According to this: http://www.cplusplus.com/reference/cstdio/printf/ the standard seems to be to return the number of characters written, or a negative number if unsuccessful. If `tfp_format()` keeps track of that already, we could just change its return type too, and use that. Though I don't know how useful this information is, so returning `1` for now might be good enough.

Reply:
It would be a more invasive change, things like `uli2a` would need to be updated to return a count of written characters, then `tfp_format` would need keep the tally. `tfp_sprintf` could potentially return `strlen` of the output buffer if it wants to be slightly more accurate. To be 100% compliant, we would need to do all the extra error handling.

At the moment, I kinda just want my board to compile... though I do see long term value in it being feature complete.

Reply:
I wonder if we can use this fork of tinyprintf instead: https://github.com/cjlano/tinyprintf
It seems to have fixed a few bugs and added more functions, plus `tfp_sprintf()` returns the number of characters already!

Reply:
Still has the tfp_printf void issue, and we would need to port the binary support into it. Worth considering, plus theres alternatives like <https://github.com/mpaland/printf> which tick both non void functions and binary support.